lib/engine/game/g_1862/step/merge.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1862
+      module Step
+        class Merge < Engine::Step::Base
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity
+            return [] if entity.receivership? || @game.skip_round[entity]
+
+            return ['choose'] if @merging
+
+            %w[merge pass]
+          end
+
+          def log_skip(entity)
+            super unless @game.skip_round[entity]
+          end
+
+          def auto_actions(entity)
+            return super if @merging
+
+            return [Engine::Action::Pass.new(entity)] if mergeable_candidates(entity).empty?
+
+            super
+          end
+
+          def merge_name
+            'Merge'
+          end
+
+          def merger_auto_pass_entity
+            current_entity
+          end
+
+          def can_merge?(entity)
+            mergeable_candidates(entity).any?
+          end
+
+          def description
+            return 'Choose Survivor' if @merging
+
+            'Merge with Major Corporation'
+          end
+
+          def process_merge(action)
+            @merging = [action.entity, action.corporation]
+            @log << "#{@merging.first.name} and #{@merging.last.name} will merge"
+          end
+
+          def process_choose(action)
+            if action.choice == :first
+              @log << "#{@merging.last.name} (non-survivor) will merge into #{@merging.first.name} (survivor)"
+              survivor = @merging.first
+              nonsurvivor = @merging.last
+            else
+              @log << "#{@merging.first.name} (non-survivor) will merge into #{@merging.last.name} (survivor)"
+              survivor = @merging.last
+              nonsurvivor = @merging.first
+            end
+            @game.start_merge(action.entity, survivor, nonsurvivor, :merge)
+            @merging = nil
+          end
+
+          def mergeable_type(corporation)
+            "Corporations that can merge with #{corporation.name}"
+          end
+
+          def mergeable_candidates(corporation)
+            # Mergeable candidates must be connected by track
+            parts = @game.graph.connected_nodes(corporation).keys
+              .reject { |n| @game.class::LONDON_TOKEN_HEXES.include?(n.hex.id) }
```

Comment:
If the corporation has a token in London, will connected_nodes return cities connected to London that aren't connected by other tokens of that corporation? If so, is it legal to merge via your London token?

Reply:
I don't think so since there aren't any paths into the London cities, but I'll think about this.

Reply:
Actually, that's why I reject the LONDON_TOKEN_HEXES

Reply:
Yeah, I saw that; however, it happens after the conencted_nodes call, so I wasn't sure if the nodes that are *only* connected to London and also in the list and should be removed.

Reply:
It should work. The connected_nodes call returns a list of all nodes that corporation connects to. Then we check to see if any other corps have a token in that node. I filter out the London nodes, and since the London cities can't connect to any other part of the graph (since there are no paths connecting to those cities), it should work.